### PR TITLE
Merge `mxbox` into `gluon.data`

### PR DESCRIPTION
**Pull request by Lyken17**
Refer to https://github.com/apache/incubator-mxnet/issues/7376#issuecomment-330006540

What I did:

1. Add `transform` utils to `gluon.data`.
    * Since `transform` handles `PIL` image by default, dependency `Pillow` is added to `setup.py`
2. Split vision dataset into separate files.

### CODE REVIEW

**Comment by Lyken17** [comment]
What are left:

1. An efficient multi-process loader
I notice there is already one single process loader. `MXBox` has a multi-thread loader, however due to the GIL of python, multi-thread does not help much when loading large image. Meanwhile, fetching large data from queue will cause overhead in multi-process. The best solution I have seen so far is the `pin_memory` in `pytorch.Tensor` -- which passes a reference between processes instead of copying. 

2. CIFAR100 / SVHN  more dataesets

**Comment by szha** [comment]
CIFAR100 is being added in #7892

File: python/mxnet/gluon/data/transforms/general.py [hunk 1]
```diff
@@ -0,0 +1,254 @@
+from __future__ import division
+
+import math
+import random
+
+from PIL import Image, ImageOps
```
Comment (piiswrong) [inline python/mxnet/gluon/data/transforms/general.py]: we don't depend on PIL. Use mx.image

File: python/mxnet/gluon/data/__init__.py [hunk 2]
```diff
@@ -25,4 +25,7 @@
 
 from .dataloader import *
 
-from . import vision
+# from . import vision
+from . import vision_dataset as vision
```
Comment (piiswrong) [inline python/mxnet/gluon/data/__init__.py]: create a sub folder named vision and move the files in there.

File: python/mxnet/gluon/data/transforms/numpyTool.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
+from __future__ import division
+
+import math
+import random
+from PIL import Image, ImageOps
+import PIL
+
+try:
+    import accimage
```
Comment (piiswrong) [inline python/mxnet/gluon/data/transforms/numpyTool.py]: don't use this.

File: python/mxnet/gluon/data/transforms/mxnetTool.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+try:
+    import accimage
+except ImportError:
+    accimage = None
+
+import numpy as np
+import numbers
+import types
+import collections
+import mxnet as mx
```
Comment (piiswrong) [inline python/mxnet/gluon/data/transforms/mxnetTool.py]: don't import mxnet inside mxnet.

Comment (Lyken17) [reply python/mxnet/gluon/data/transforms/mxnetTool.py]: The transformation needs some operators inside `mx.nd`. So should I use relative import like `from ..... import mxnet as mx` ?

Comment (piiswrong) [reply python/mxnet/gluon/data/transforms/mxnetTool.py]: from ... import ndarray

**Comment by piiswrong** [comment]
closing due to inactive. Please see the vision branch.

**Pull request closed without merging**